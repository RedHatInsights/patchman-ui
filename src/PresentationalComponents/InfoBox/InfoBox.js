import { Text, TextContent, TextVariants, Bullseye, Split, SplitItem } from '@patternfly/react-core';
import propTypes from 'prop-types';
import React from 'react';
import WithLoader, { WithLoaderVariants } from '../WithLoader/WithLoader';
import './InfoBox.scss';

const InfoBox = ({ title, text, isLoading, content, color }) => {
    return (
        <Split className="infobox" hasGutter>
            <WithLoader
                variant={WithLoaderVariants.skeleton}
                loading={isLoading}
                size="lg"
            >
                <SplitItem style={{ backgroundColor: color }}>
                    <Bullseye>{content}</Bullseye>
                </SplitItem>
                <SplitItem isFilled>
                    <TextContent>
                        <Text component={TextVariants.h6}>{title}</Text>
                        <Text component={TextVariants.p}>{text}</Text>
                    </TextContent>
                </SplitItem>
            </WithLoader>
        </Split>
    );
};

InfoBox.propTypes = {
    title: propTypes.string,
    text: propTypes.any,
    isLoading: propTypes.bool,
    content: propTypes.any,
    color: propTypes.string
};

export default InfoBox;
