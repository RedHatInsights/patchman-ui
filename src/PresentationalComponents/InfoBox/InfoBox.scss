.infobox {
    background-color: var(--pf-v5-global--BackgroundColor--150);
    border: var(--pf-v5-global--BorderColor--300) var(--pf-v5-global--BorderWidth--sm)
        solid;
    height: 65px;
    > .pf-v5-l-split__item:first-child {
        width: 65px;
        color: black;
        background-color: var(--pf-v5-global--warning-color--100);
        > div.pf-v5-l-bullseye svg {
            color: white;
        }
    }
    .pf-v5-c-content h6 {
        margin: 0;
    }
    > .pf-v5-l-split__item:nth-child(2) {
        display: flex;
        justify-content: flex-start;
        align-items: center;
    }
}
