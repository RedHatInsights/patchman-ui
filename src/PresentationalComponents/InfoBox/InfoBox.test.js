import InfoBox from './InfoBox';
import toJson from 'enzyme-to-json';

const props = {
    title: 'testTitle',
    text: 'testText',
    isLoading: false,
    content: 'testContent',
    color: 'testColor'
};

describe('InfoBox', () => {
    it('Should match to snapshots', () => {
        const wrapper = shallow(<InfoBox {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
