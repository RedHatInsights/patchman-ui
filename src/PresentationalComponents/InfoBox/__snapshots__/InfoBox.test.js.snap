// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox Should match to snapshots 1`] = `
<Split
  className="infobox"
  hasGutter={true}
>
  <WithLoader
    loading={false}
    size="lg"
    variant="skeleton"
  >
    <SplitItem
      style={
        Object {
          "backgroundColor": "testColor",
        }
      }
    >
      <Bullseye>
        testContent
      </Bullseye>
    </SplitItem>
    <SplitItem
      isFilled={true}
    >
      <TextContent>
        <Text
          component="h6"
        >
          testTitle
        </Text>
        <Text
          component="p"
        >
          testText
        </Text>
      </TextContent>
    </SplitItem>
  </WithLoader>
</Split>
`;
